### DISCUSSION TITLE
UseHttpMetrics() equivalent in Prometheus.Client

### DISCUSSION THREAD

**Discussion by cscoon** [PrometheusClientNet/Prometheus.Client#187] [General]
We are migrating our applications from Prometheus-net to Prometheus.Client.

Prometheus-net had an extension method to IApplicationBuilder called `UseHttpMetrics()`.  This method added several metrics regarding each HTTP request to the scraping endpoint.  According to their source code, this method added three pieces of middleware: `HttpInProgressMiddleware`, `HttpRequestCountMiddleware`, `HttpRequestDurationMiddleware`.

Prometheus.Client.HttpRequestDurations has a method named `UsePrometheusRequestDurations(...)`, but that method seems to only add `HttpRequestDurationsMiddleware` and the `HttpRequestDurationsMiddleware` source code is a part of the Prometheus.Client.HttpRequestDurations project, so it must not be the same one Prometheus-net is using.

Does Prometheus.Client have an equivalent to `UseHttpMetrics()`?

**Comment by phnx47** [comment]
@cscoon [prom-client-httprequestdurations](https://github.com/prom-client-net/prom-client-httprequestdurations) all about HTTP metrics here. You can count your requests and calculate durations.

Yes, it is not the same implementation. You can create issues or PRs there with additional features.

**Comment by sanych-sun** [comment]
@cscoon Could you please describe the missing functionality so we can consider if it worth to implement it or propose some workaround?

**Comment by cscoon** [comment]
I would need to have a complete list of all metrics that are emitted as a result of UseHttpMetrics() which I don't have.  Based on the naming of the middleware, I would presume it to be all metrics associated with Prometheus-net's `HttpInProgressMiddleware` and `HttpRequestCountMiddleware`

**Reply by sanych-sun** [reply]
I do not have it either :) We can start from another point: we already have [Prometheus.Client.HttpRequestDurations](https://github.com/prom-client-net/prom-client-httprequestdurations) on board. This library let you to create a histogram to capture http requests duration and btw have the request count (as histogram has sum).

So probably you can describe which more metric do you need (probably by investigating the dashboards on your Prometheus) and we can discuss if it makes sense to implement that metrics on our side.
